pkg/mkverification-raw-efi/initramfs-init.patch

```diff
@@ -0,0 +1,116 @@
++        for i in 0 1 2; do
++                ebegin "Attempt $i to find and mount boot media"
++                MEDIA_ID=$(grep -l "$KOPT_find_boot" /media/*/boot/.uuid 2>/dev/null)
++                if [ -n "$MEDIA_ID" ]; then
++                        mkdir -p /media/boot
++                        mount --bind "/media/$(echo "$MEDIA_ID" | cut -f3 -d/)" /media/boot
++                        break
++                fi
++                sleep $(( i * 30 ))
++                nlplug-findfs $cryptopts -p /sbin/mdev ${KOPT_debug_init:+-d} \
++                   ${KOPT_usbdelay:+-t $(( $KOPT_usbdelay * 1000 ))} \
++                   -n -b $repofile -a /tmp/apkovls
++                eend $?
++        done
++
++        # if we didn't find anything, but were asked to -- treat it
++        # as an error condition (it maybe transient, but it needs to
++        # be corrected
++        if [ -z "$MEDIA_ID" ]; then
++                echo "Failed to identify boot media. Try to re-run nlplug-findfs manually to see what's wrong:"
++                echo "  nlplug-findfs -p /sbin/mdev -d -t 30000 -n  -n -b $repofile -a /tmp/apkovls"
++                echo "once you find boot device, run:"
++                echo "  mount --bind /media/XXX /media/boot"
++                echo "and then exit the shell."
++                sh
++        fi
++fi
++
+ # check if root=... was set
+ if [ -n "$KOPT_root" ]; then
+ 	if [ "$SINGLEMODE" = "yes" ]; then
+@@ -504,9 +538,24 @@
+ 	fi
+ 
+ 	ebegin "Mounting root"
+-	nlplug-findfs $cryptopts -p /sbin/mdev ${KOPT_debug_init:+-d} \
+-		$KOPT_root
++        if [ -f "$KOPT_root" ]; then
++                LOOP_IMG=$(realpath "$KOPT_root")
++                # workingaround linux kernel's desire to lump the entire
++                # set of initrd images into /initrd.image
++                if [ "$LOOP_IMG" = /initrd.image ]; then
++                        OFFSET=$(LANG=C grep -obUaP hsqs /initrd.image|cut -f1 -d:|head -1)
++                        if [ -n "$OFFSET" ]; then
++                                LOSETUP_EXTRA_OPTS="-o$OFFSET"
++                        fi
++                fi
+
++                KOPT_root=$(losetup -f)
++                losetup $LOSETUP_EXTRA_OPTS -r -f "$LOOP_IMG"
++        else
++                nlplug-findfs $cryptopts -p /sbin/mdev ${KOPT_debug_init:+-d} \
++                        $KOPT_root
++        fi
++
+ 	if echo "$KOPT_modules $rootfstype" | grep -qw btrfs; then
+ 		/sbin/btrfs device scan >/dev/null || \
+ 			echo "Failed to scan devices for btrfs filesystem."
+@@ -540,13 +555,15 @@
+ 		mount ${KOPT_rootfstype:+-t $KOPT_rootfstype} -o $rootflags \
+ 			$KOPT_root /media/root-ro
+ 		# Mount writable overlay tmpfs
+-		overlaytmpfsflags="mode=0755,${KOPT_overlaytmpfsflags:+$KOPT_overlaytmpfsflags,}rw"
+-		mount -t tmpfs -o $overlaytmpfsflags root-tmpfs /media/root-rw
++		# overlaytmpfsflags="mode=0755,${KOPT_overlaytmpfsflags:+$KOPT_overlaytmpfsflags,}rw"
```

Comment:
If you don't need these lines, just remove them (along with the above comment)....

Reply:
I am trying to understand. This patch instructs to remove lines 93 and 94 and insert line 95. Are you saying we should also remove line 92?